flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SynchronousCheckpointTest.java

```diff
@@ -76,24 +75,6 @@ public void setupTestEnvironment() throws InterruptedException {
         assertThat(eventQueue.take(), is(Event.TASK_INITIALIZED));
     }
 
-    @Test(timeout = 20_000)
-    public void synchronousCheckpointBlocksUntilNotificationForCorrectCheckpointComes()
```

Comment:
This depends on a shady `getSynchronousSavepointId` and is covered by tests in `StreamTaskFinalCheckpointsTest`.

The reason why it does not work is that we do not clear the `synchronousSavepointId` internally, because it is not necessary. Previously, clearing the flag was used to exit the synchronous loop. Now, without a synchronous loop, we simply exit.